lib/elixir/lib/string.ex

```diff
@@ -1755,8 +1755,8 @@ defmodule String do
     end
   end
 
-  @compile {:inline, decompose: 1}
-  defp decompose(string) do
+  @compile {:inline, dissect: 1}
+  defp dissect(string) do
```

Comment:
Wdyt about calling this something more straightforward, like `chars_and_their_count`? It's a private function and it's used in very few places, so the long name shouldn't be a problem. Maybe I'm missing something, but neither `dissect` nor `decompose` make it easy for me to see what this function does.